### ISSUE TITLE
`sudo dnf system-upgrade reboot ` ran into boot loop

### ISSUE THREAD

**Issue by yejingchen** [rpm-software-management/dnf-plugin-system-upgrade#35]
Hi, I have an issue upgrading my fedora 22 to 23. I installed `dnf-plugin-system-upgrade`, `python2-dnf-plugin-system-upgrade` and `python3-dnf-plugin-system-upgrade`, and ran a `sudo dnf-3 update` to make sure everything is up-to-date. (I used `dnf-3` because under Chinese locale (`zh_CN.utf8`), the python2 version of the plugin makes dnf unable to ran any command even `dnf help`)

The download command (`sudo dnf-3 system-upgrade download --releasever=23 --best`) ran smoothly, but the reboot command didn't function as it should. After reboot it booted into f22 kernel, displayed the fedora logo, then rebooted again, and again, and again. During the reboots I saw several lines like 'Starting System Upgrade' 'Started System Upgrade'. I followed the instructions in README.md, deleted `/system-update`, and came here to file a bug.

I upload my `/var/log/boot.log` [here](http://paste.ubuntu.com/13122555/), hope it will be useful.

**Comment by kparal** [comment]
Please include `/var/log/dnf.log` and the output of `dnf system-upgrade log -1`, thanks.

The problem might be that you used python3 to download the packages, but not to upgrade the packages. I advise to try to change the system language to english and then try again.

**Comment by yejingchen** [comment]
I think I have to make an apology... Because I have just upgraded to 23 by switching system language to English before I make backups of the two files, which means `/var/log/dnf.log` and `dnf system-upgrade log -1` might be no longer the ones containing useful information. `dnf system-upgrade log` now says there is only one log, so the last one was overridden somehow.

But given the comments (4, 5, 6) of the [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1278031) you filed on Red Hat Bugzilla, I suspect this may be a bug in i18n of `dnf`, althought I actually don't know much about it.

**Comment by kparal** [comment]
Hmm, I assume `dnf system-upgrade log -1` does not see the original upgrade attempt, because it was not started properly. You can definitely still access it, using `journalctl --list-boots`, finding the proper one, and then displaying it by `journalctl -a -b NUMBER`. (e.g. `-5` for 5 boots earlier). But since this problem is easy to reproduce, I suppose you don't need to spend too much time on this, the developers should be able to gather necessary information even without your files.

**Comment by yejingchen** [comment]
I found the one containing the dnf traceback I met when running dnf upgrade command, and I pasted it [here](http://paste.ubuntu.com/13136381/). (Not so sure whether ubuntu pastebin is an appropriate place to upload these log files containing 5000+ lines. If you have  a better choice, tell me please :)

**Comment by kparal** [comment]
That's not the right log, this is the session where you executed `dnf system-upgrade reboot`, we need the following one (where the upgrade was supposed to happen). As for pasting long text files, github supports a similar service called Gist:
https://help.github.com/articles/about-gists/

**Comment by yejingchen** [comment]
Here's the following log containing 20000+ lines:
https://gist.github.com/yejingchen/47aa4aea6664d8f49c8a

Sorry for the delay :)

**Comment by maekitalo** [comment]
I have the same problem. I did the very same. I strictly followed the upgrade recomendation from the wiki. One problem was, that I had to run every dnf command with LANG=C prepended. But unfortunately I did not change the language of my system. It is german. Now it keeps rebooting. How can I access the log when I do not get a shell?

**Comment by maekitalo** [comment]
Hi again,

I just found the answer to my question: https://github.com/rpm-software-management/dnf-plugin-system-upgrade#in-case-of-boot-problems

**Comment by wgwoods** [comment]
So the _root cause_ of your reboot loops (i.e. the `LANG` stuff) should be fixed in commit 4bbbf1ec0c3f387249d30dd103ff4d002c17f574.

Furthermore, commit 7e91ff543ff98ad38646c7dacad23aa389a00e17 (from PR #44) should prevent any future problems from causing these kinds of reboot loops. So I think we've got this issue fixed.

**Comment by maekitalo** [comment]
Thank you for the fix.

I solved the issue. These were the steps I did:
- to break the reboot loop I followed the instructions in the above link. In the grub boot menu pressed 'e' and added this rd.break in the parameters line starting with "linux"
- after reboot I had my old fedora back. In my case it was 21.
- I set the system locale using "localectrl set-locale LANG=C"
- running now "dnf system-upgrade reboot" updated my machine

On my other systems I started the whole process by first setting the system locale. Also when running a root shell (using sudo bash) I set LANG=C before running dnf commands.
The update process is now running and hopefully finishes in the next hours.
